### PR TITLE
update kubernetes dependency to v1.23

### PR DESCRIPTION
**Pull request by ramineni**
**What this PR does / why we need it**:

**Which issue this PR fixes(if applicable)**:
fixes #

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jichenjc** [comment]
```
Summarizing 3 Failures:

[Fail] [sig-storage] [cinder-csi-e2e] CSI Volumes [Driver: cinder.csi.openstack.org] [Testpattern: Generic Ephemeral-volume (block volmode) (late-binding)] ephemeral [It] should create read/write inline ephemeral volume 
/root/pkg/mod/k8s.io/kubernetes@v1.23.0/test/e2e/storage/testsuites/ephemeral.go:428

[Fail] [sig-storage] [cinder-csi-e2e] CSI Volumes [Driver: cinder.csi.openstack.org] [Testpattern: Generic Ephemeral-volume (block volmode) (late-binding)] ephemeral [It] should support multiple inline ephemeral volumes 
/root/pkg/mod/k8s.io/kubernetes@v1.23.0/test/e2e/storage/testsuites/ephemeral.go:428

[Fail] [sig-storage] [cinder-csi-e2e] CSI Volumes [Driver: cinder.csi.openstack.org] [Testpattern: Generic Ephemeral-volume (block volmode) (late-binding)] ephemeral [It] should support two pods which have the same volume definition 
/root/pkg/mod/k8s.io/kubernetes@v1.23.0/test/e2e/storage/testsuites/ephemeral.go:428
```

from e2e seems some changes are needed , might be worthy to dig into whether some new features added/removal since 1.23...

**Comment by ramineni** [comment]
> ```
> Summarizing 3 Failures:
> 
> [Fail] [sig-storage] [cinder-csi-e2e] CSI Volumes [Driver: cinder.csi.openstack.org] [Testpattern: Generic Ephemeral-volume (block volmode) (late-binding)] ephemeral [It] should create read/write inline ephemeral volume 
> /root/pkg/mod/k8s.io/kubernetes@v1.23.0/test/e2e/storage/testsuites/ephemeral.go:428
> 
> [Fail] [sig-storage] [cinder-csi-e2e] CSI Volumes [Driver: cinder.csi.openstack.org] [Testpattern: Generic Ephemeral-volume (block volmode) (late-binding)] ephemeral [It] should support multiple inline ephemeral volumes 
> /root/pkg/mod/k8s.io/kubernetes@v1.23.0/test/e2e/storage/testsuites/ephemeral.go:428
> 
> [Fail] [sig-storage] [cinder-csi-e2e] CSI Volumes [Driver: cinder.csi.openstack.org] [Testpattern: Generic Ephemeral-volume (block volmode) (late-binding)] ephemeral [It] should support two pods which have the same volume definition 
> /root/pkg/mod/k8s.io/kubernetes@v1.23.0/test/e2e/storage/testsuites/ephemeral.go:428
> ```
> 
> from e2e seems some changes are needed , might be worthy to dig into whether some new features added/removal since 1.23...

Thanks. I dont see much of changes . Investigating if anything amiss

**Comment by jichenjc** [comment]
```
commit a40d2eb18c7a256f33b4839a3a46172321128aa8
Author: Patrick Ohly <patrick.ohly@intel.com>
Date:   Thu Oct 14 18:01:09 2021 +0200

    storage validation: accept generic ephemeral volumes as volume device

    Raw block devices are possible with generic ephemeral volumes, so rejecting a
    pod with that combination is wrong.

```

seems this is the main error
```
Message: "Pod \"inline-volume-tester-5rw6d\" is invalid: spec.containers[0].volumeDevices[0].name: Invalid value: \"my-volume-0\": can only use volume source type of PersistentVolumeClaim for block mode",
```
the above PR is latest change of test between 1.22/23, not sure that's the reason..

**Comment by ramineni** [comment]
> ```
> commit a40d2eb18c7a256f33b4839a3a46172321128aa8
> Author: Patrick Ohly <patrick.ohly@intel.com>
> Date:   Thu Oct 14 18:01:09 2021 +0200
> 
>     storage validation: accept generic ephemeral volumes as volume device
> 
>     Raw block devices are possible with generic ephemeral volumes, so rejecting a
>     pod with that combination is wrong.
> ```
> 
> seems this is the main error
> 
> ```
> Message: "Pod \"inline-volume-tester-5rw6d\" is invalid: spec.containers[0].volumeDevices[0].name: Invalid value: \"my-volume-0\": can only use volume source type of PersistentVolumeClaim for block mode",
> ```
> 
> the above PR is latest change of test between 1.22/23, not sure that's the reason..

@jichenjc yes, those tests are added in 1.23 only and yes its failing with above error. But we dont pass those values from driver , and in the code it seems to be creating fine. https://github.com/kubernetes/kubernetes/blob/ab69524f795c42094a6630298ff53f3c3ebab7f4/test/e2e/storage/framework/volume_resource.go#L136

@jichenjc Could you please also look at it if anything amiss from the driver side

**Comment by jichenjc** [comment]
@ramineni yes, I am also checking it :) let's exchange info in case anything new found

**Comment by ramineni** [comment]
@jichenjc @lingxiankong looks like issue to be k3s doesnt support to install 1.23 yet I suppose.

@lingxiankong could you please check?

**Comment by jichenjc** [comment]
>>>@jichenjc @lingxiankong looks like issue to be k3s doesnt support to install 1.23 yet I suppose.

oh.. https://github.com/k3s-io/k3s/releases
don't have 1.23 yet, maybe we can keep an eye and wait for a while

**Comment by ramineni** [comment]
/override openlab/cloud-provider-openstack-multinode-csi-migration-test
/override openlab/cloud-provider-openstack-acceptance-test-keystone-authentication-authorization
/override openlab/cloud-provider-openstack-acceptance-test-e2e-conformance-stable-branch-v1.21
/override openlab/cloud-provider-openstack-acceptance-test-e2e-conformance-stable-branch-v1.20
openlab/cloud-provider-openstack-acceptance-test-e2e-conformance-stable-branch-v1.19

**Comment by ramineni** [comment]
/override openlab/cloud-provider-openstack-acceptance-test-e2e-conformance-stable-branch-v1.19
/override openlab/cloud-provider-openstack-acceptance-test-e2e-conformance

**Comment by ramineni** [comment]
@gman0 @gouthampacha could you check manila failure if available. Thanks

**Comment by gman0** [comment]
@ramineni thanks for pinging, but I'm on a vacation. I can take a look later next week though.

**Comment by gman0** [comment]
@ramineni do you know if there any cmd flag changes in klog? It's failing on log verbosity setting:

```
Error: unknown shorthand flag: 'v' in -v=2
```

**Comment by sdodson** [comment]
/hold
This missed feature freeze. We need to discuss whether or not these make sense to land after feature freeze.

**Comment by jichenjc** [comment]
>>>We need to discuss whether or not these make sense to land after feature freeze.

@sdodson what's the feature freeze ? CPO is still catch up with 1.23 k8s release so not sure the meaning of feature freeze, thanks

**Comment by jichenjc** [comment]
>>>Error: unknown shorthand flag: 'v' in -v=2

is there special usage of manila plugin? I think OCCM/CSI cinder both use -v in various places ?

**Comment by sdodson** [comment]
/hold cancel
Sorry I didn't realize this was an upstream repo

**Comment by ramineni** [comment]
> unknown shorthand flag

@gman0 i dont understand , where do you see this error. I cant seem to find error in logs

**Comment by ramineni** [comment]
@gman0 Did u test manila plugin in local setup with upgrade to 1.23 kk . This is blocker for release .

**Comment by gman0** [comment]
Yes I tested this PR (in a v1.23 cluster too) and `-v` that's being passed from the Helm chart is causing klog to fail -- it doesn't seem to recognize this option anymore.

**Comment by mandre** [comment]
> Yes I tested this PR (in a v1.23 cluster too) and `-v` that's being passed from the Helm chart is causing klog to fail -- it doesn't seem to recognize this option anymore.

Regarding the `-v` issue, you'll have to use the new `cli.Run()` function from "k8s.io/component-base/cli".
See https://github.com/kubernetes/kubernetes/pull/105076/.

**Comment by gman0** [comment]
@mandre thank you for this info! Most of the projects in this repo needs an update it seems.

**Comment by mandre** [comment]
> Most of the projects in this repo need an update it seems.

That would be just `cmd/openstack-cloud-controller-manager/main.go` if I'm not mistaken. We should update similar to what was done in https://github.com/kubernetes/kubernetes/pull/105076/files#diff-553e5a7eabe071db91dd2467db6a7fb68bcddf2506ebbf7fccb9335ea632fd94 for example.

**Comment by gman0** [comment]
@mandre It looks like it's actually all projects except occm :) @ramineni could you please allow write access to this PR so I can push commits here?

This is what it looks like with patches applied https://gist.github.com/gman0/0256a99b2a7678a375ec581f09131e01

**Comment by ramineni** [comment]
> @mandre It looks like it's actually all projects except occm :) @ramineni could you please allow write access to this PR so I can push commits here?
> 
> This is what it looks like with patches applied https://gist.github.com/gman0/0256a99b2a7678a375ec581f09131e01

I dont see option to give write access . Please push another PR if you have changes ready , I can close this one

**Comment by gman0** [comment]
@ramineni according to https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork it should be right here on the PR, in the right vertical menu/bar thing.

**Comment by ramineni** [comment]
./test openstack-cloud-keystone-authentication-authorization-test

File: go.mod [hunk 1]
```diff
@@ -1,81 +1,182 @@
...
-	k8s.io/apiserver v0.22.0
-	k8s.io/client-go v0.22.0
-	k8s.io/cloud-provider v0.22.0
-	k8s.io/component-base v0.22.0
-	k8s.io/klog v1.0.0 // indirect
-	k8s.io/klog/v2 v2.9.0
-	k8s.io/kubernetes v1.22.0
-	k8s.io/mount-utils v0.22.0
-	k8s.io/utils v0.0.0-20210707171843-4b05e18ac7d9
+	k8s.io/api v0.23.0
```
Comment (zetaab) [inline go.mod:32]: 0.23.1 is out already

**Comment by zetaab** [comment]
@ramineni could you update k8s.io deps to 1.23.1? after that I can approve this PR but need to know what is status of those failing tests. Should those work or is it ok to skip

**Comment by ramineni** [comment]
@jichenjc openstack-cloud-keystone-authentication-authorization-test , this test is not up right? ill override it?

**Comment by ramineni** [comment]
/override openlab/cloud-provider-openstack-acceptance-test-e2e-conformance

**Comment by ramineni** [comment]
/override openlab/cloud-provider-openstack-acceptance-test-e2e-conformance-stable-branch-v1.19
/override openlab/cloud-provider-openstack-acceptance-test-e2e-conformance-stable-branch-v1.20
/override openlab/cloud-provider-openstack-acceptance-test-e2e-conformance-stable-branch-v1.21
/override openlab/cloud-provider-openstack-multinode-csi-migration-test

**Comment by ramineni** [comment]
/override openlab/cloud-provider-openstack-acceptance-test-keystone-authentication-authorization

**Comment by jichenjc** [comment]
>>>ill override it?

yes , please

**Comment by ramineni** [comment]
/override openstack-cloud-keystone-authentication-authorization-test

**Pull request merged**